components/Button.tsx

```diff
@@ -104,15 +107,18 @@ const Button = React.forwardRef<
         >
           {children}
         </span>
-        <span
-          css={(theme) => css`
-            position: absolute;
-            visibility: ${shouldShowLoading ? 'visible' : 'hidden'};
-            bottom: 1px;
-          `}
-        >
-          <Spinner height={20} width={20} />
-        </span>
+
+        {isAsync && (
+          <span
```

Comment:
This corrects a weird behaviour in which non async buttons would have a ghost animated spinner in the DOM tree, which is unnecessary, but may introduce performance drag in older devices.